File: src/main/java/szewek/mcflux/recipes/RecipeFluxGen.java
package szewek.mcflux.recipes;

public final class RecipeFluxGen {
	public final short factor, usage;

	RecipeFluxGen(int f, int u) {
		factor = (short) f;
		usage = (short) u;
	}

	@Override public int hashCode() {
		return ((int) factor << 16) + usage;
	}

	@Override public boolean equals(Object obj) {
		return obj != null && obj instanceof RecipeFluxGen && obj.hashCode() == hashCode();
	}
}

File: src/main/java/szewek/mcflux/recipes/RecipeFluxGen.kt
package szewek.mcflux.recipes

class RecipeFluxGen internal constructor(f: Int, u: Int) {
	val factor = f.toShort()
	val usage = u.toShort()

	override fun hashCode(): Int {
		return (factor.toInt() shl 16) + usage
	}

	override fun equals(other: Any?): Boolean {
		return other != null && other is RecipeFluxGen && other.hashCode() == hashCode()
	}
}
